examples/client/client.c

```diff
@@ -3089,8 +3090,53 @@ THREAD_RETURN WOLFSSL_THREAD client_test(void* args)
 #ifdef HAVE_SECURE_RENEGOTIATION
     if (scr && forceScr) {
         if (nonBlocking) {
-            printf("not doing secure renegotiation on example with"
-                   " nonblocking yet\n");
+            if (!resumeScr) {
+                if ((ret = wolfSSL_Rehandshake(ssl)) != WOLFSSL_SUCCESS) {
+                    err = wolfSSL_get_error(ssl, 0);
+                    if (err == WOLFSSL_ERROR_WANT_READ ||
+                            err == WOLFSSL_ERROR_WANT_WRITE) {
+                        (void)ClientWrite(ssl,
+                                "msg sent during renegotiation",
```

Comment:
A) Now checking return code
B) Since the RFC requires that processing APP DATA during a renegotiation is supported, this is a way to make sure that we don't lose this support. I can make it an option but the command line options look to be pretty full already.

Reply:
Dong get me wrong, I like the idea of testing App Data during a renegotiation but I would like to avoid ALWAYS ONLY testing App Data during a renegotiation vs the way more common case of not having App Data during.  See if that can easily be added, thanks.

Reply:
App data during a renegotiation is only tested when using non-blocking sockets. I see that the '-9' command line option. Should I use that or add a value for -i (like './examples/client/client -i test-app-data')?

Reply:
I don't have an opinion on the option name / title.  Always testing APP Data during non-blocking is not good either.  We HAVE to test the most common cases as well as exceptional ones.  Not JUST exceptional ones.

Reply:
I added an option to `-i` where if the value of this argument is `scr-app-data` then it sends app data. I also patched `mygetopt` so that it will detect if the value being passed to an argument is the next argument.